test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SampleControllerSpec.scala

```diff
@@ -0,0 +1,245 @@
+  )
+
+  private def controller(requestCase: Case, permission: Set[Permission]) = new SampleController(
+    new RequestActionsWithPermissions(permission, c = requestCase), casesService, messageApi, appConfig
+  )
+
+  private implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def afterEach(): Unit = {
+    super.afterEach()
+    Mockito.reset(casesService)
+  }
+
+  "Sample controller" should {
+
+    "return 200 OK and HTML content type - For CANCELLED Case" in {
+      val c = aCase(withStatus(CaseStatus.CANCELLED), withDecision())
+
+      val result = await(controller(c).sampleDetails("reference")(fakeRequest))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("sample-status-heading")
+    }
+
+    "return 200 OK and HTML content type - For COMPLETED Case" in {
+      val c = aCase(withStatus(CaseStatus.COMPLETED), withDecision())
+
+      val result = await(controller(c).sampleDetails("reference")(fakeRequest))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("sample-status-heading")
+    }
+
+    "return 200 OK and HTML content type - For OPEN Case" in {
+      val c = aCase(withStatus(CaseStatus.OPEN), withDecision())
+
+      val result = await(controller(c).sampleDetails("reference")(fakeRequest))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("sample-status-heading")
+    }
+
+  }
+
+  "Sample - Choose Status" should {
+
+    "return 200 OK and HTML content type - For CANCELLED Case" in {
+      val c = aCase(withStatus(CaseStatus.CANCELLED), withDecision(appeal = Some(Appeal(AppealStatus.IN_PROGRESS))))
+
+      val result = await(controller(c).chooseStatus("reference")(newFakeGETRequestWithCSRF(fakeApplication)))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("change_sample_status-heading")
+    }
+
+    "return 200 OK and HTML content type - For COMPLETED Case" in {
+      val c = aCase(withStatus(CaseStatus.COMPLETED), withDecision())
+
+      val result = await(controller(c).chooseStatus("reference")(newFakeGETRequestWithCSRF(fakeApplication)))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("change_sample_status-heading")
+    }
+
+    "return 200 OK and HTML content type - For OPEN Case" in {
+      val c = aCase(withStatus(CaseStatus.OPEN), withDecision())
```

Comment:
Same for these three tests - is case status relevant?

Reply:
Have updated this one and removed irrelevant repeats